crates/uv-dispatch/src/lib.rs

```diff
@@ -315,34 +307,20 @@ impl<'a> BuildContext for BuildDispatch<'a> {
         dist: Option<&'data SourceDist>,
         build_kind: BuildKind,
     ) -> Result<SourceBuild> {
-        match self.no_build {
-            NoBuild::All => debug_assert!(
-                match self.no_binary {
-                    // Allow `all` to be overridden by specific binary exclusions
-                    NoBinary::Packages(packages) => {
-                        if let Some(dist) = dist {
-                            packages.contains(dist.name())
-                        } else {
-                            false
-                        }
-                    }
-                    _ => false,
-                } || matches!(build_kind, BuildKind::Editable),
-                "Only editable builds are exempt from 'no build' checks"
-            ),
-            NoBuild::None => {}
-            NoBuild::Packages(packages) => {
-                // We can only prevent builds by name for packages with names. For editable
-                // packages and unnamed requirements, we can't prevent the build.
-                if let Some(dist) = dist {
-                    if packages.contains(dist.name()) {
-                        bail!(
-                            "Building source distributions for {} is disabled",
-                            dist.name()
-                        );
-                    }
-                }
+        // Note we can only prevent builds by name for packages with names. For editable
+        // packages and unnamed requirements, we can't prevent the build.
+        if self
+            .build_options
+            .no_build(dist.map(distribution_types::Name::name))
+            && !matches!(build_kind, BuildKind::Editable)
+        {
+            if let Some(dist) = dist {
+                bail!(
+                    "Building source distributions for {} is disabled",
+                    dist.name()
+                );
             }
+            bail!("Building source distributions is disabled");
         }
```

Comment:
Should this be an `else`?

Reply:
I think it's unreachable but it wasn't clear to me.

Reply:
I wrote it as an else and the compiler complained that it was redundant, it shouldn't be unreachable.

Reply:
The distribution can be `None` and we won't have a name. It's definitely reachable because I hit this error when I broke the tests :)

Reply:
I honestly might just write out `return Err` rather than the `bail` to make it clearer.

Reply:
That's fine with me, just copied this from the old code.